components/brave_rewards/resources/shared/components/vbat_notice.tsx

```diff
@@ -0,0 +1,105 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import { LocaleContext, formatMessage } from '../lib/locale_context'
+import { UserType } from '../lib/user_type'
+import { NewTabLink } from './new_tab_link'
+import { AlertIcon } from './icons/alert_icon'
+import { ArrowNextIcon } from './icons/arrow_next_icon'
+import { CloseIcon } from './icons/close_icon'
+
+import * as urls from '../lib/rewards_urls'
+import * as style from './vbat_notice.style'
+
+const dateFormatter = new Intl.DateTimeFormat(undefined, {
+  year: 'numeric',
+  month: 'long',
+  day: 'numeric',
+  hour: 'numeric'
+})
+
+export function shouldShowVBATNotice (
+  userType: UserType,
+  vbatDeadline: number | undefined
+) {
+  return (
+    userType === 'legacy-unconnected' &&
+    vbatDeadline &&
+    vbatDeadline > Date.now()
+  )
+}
+
+interface Props {
+  vbatDeadline: number | undefined
+  canConnectAccount: boolean
+  declaredCountry: string
+  onConnectAccount: () => void
+  onClose?: () => void
+}
+
+export function VBATNotice (props: Props) {
+  const { getString } = React.useContext(LocaleContext)
+  if (!props.vbatDeadline) {
+    return null
+  }
+
+  const deadline = new Date(props.vbatDeadline)
```

Comment:
Could this be moved inside `getMessages`? Maybe it doesn't matter too much...

Reply:
Done.